lib/fireblocks/api/client.go

```diff
@@ -0,0 +1,123 @@
+package fireblocks
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/golang-jwt/jwt"
+)
+
+type Client struct {
+	apiKey       string
+	clientSecret string
+	host         string
+	http         http.Client
+}
+
+const TransactionEndpoint string = "v1/transactions"
+
+func NewClient(apiKey string, clientSecret string, host string) *Client {
+	return &Client{
+		apiKey:       apiKey,       // pragma: allowlist secret
+		clientSecret: clientSecret, // pragma: allowlist secret
+		host:         host,
+	}
+}
+
+func (c *Client) sendRequest(ctx context.Context, req *http.Request, endpoint string, request []byte, nonce string) ([]byte, error) {
```

Comment:
only use pointer receivers for mutable state, all the Client fields are immutable, so no need for pointer receivers.